ui/ducks/ens.js

```diff
@@ -0,0 +1,183 @@
+            : ENS_NOT_FOUND_ON_NETWORK;
+      } else if (reason.message === 'Illegal Character for ENS.') {
+        state.error = ENS_ILLEGAL_CHARACTER;
+      } else {
+        log.error(reason);
+        state.error = ENS_UNKNOWN_ERROR;
+      }
+    },
+    processEnsResult: (state, action) => {
+      state.resolution = action.payload;
+      if (state.resolution === ZERO_ADDRESS) {
+        state.error = ENS_NO_ADDRESS_FOR_NAME;
+      }
+      if (state.resolution === ZERO_X_ERROR_ADDRESS) {
+        state.error = ENS_REGISTRATION_ERROR;
+      }
+      if (
+        isValidDomainName(state.resolution) &&
+        isConfusing(state.resolution)
+      ) {
+        state.warning = CONFUSING_ENS_ERROR;
+      }
+    },
+    enableEnsLookup: (state, action) => {
+      state.stage = 'INITIALIZED';
+      state.error = null;
+      state.resolution = null;
+      state.warning = null;
+      state.network = action.payload;
+    },
+    disableEnsLookup: (state) => {
+      state.stage = 'NO_NETWORK_SUPPORT';
+      state.error = ENS_NOT_SUPPORTED_ON_NETWORK;
+      state.warning = null;
+      state.resolution = null;
+      state.network = null;
+    },
+    resetResolution: (state) => {
+      state.resolution = null;
+      state.warning = null;
+      state.error =
+        state.stage === 'NO_NETWORK_SUPPORT'
+          ? ENS_NOT_SUPPORTED_ON_NETWORK
+          : null;
+    },
+  },
+  extraReducers: (builder) => {
+    builder.addCase(CHAIN_CHANGED, (state, action) => {
+      if (action.payload !== state.currentChainId) {
+        state.stage = 'UNINITIALIZED';
+      }
+    });
+  },
+});
+
+const { reducer, actions } = slice;
+export default reducer;
+
+const {
+  disableEnsLookup,
+  processEnsError,
+  processEnsResult,
+  enableEnsLookup,
+  resetResolution,
+} = actions;
+export { resetResolution };
+
+export function initializeEnsSlice() {
+  return async (dispatch, getState) => {
+    const state = getState();
+    const chainId = getCurrentChainId(state);
+    const network = CHAIN_ID_TO_NETWORK_ID_MAP[chainId];
+    const networkIsSupported = Boolean(networkMap[network]);
+    if (networkIsSupported) {
+      ens = new ENS({ provider: global.ethereumProvider, network });
+      await dispatch(enableEnsLookup(network));
+    } else {
+      ens = null;
+      await dispatch(disableEnsLookup());
+    }
+  };
+}
+
+export function lookupEnsName(ensName) {
+  return async (dispatch, getState) => {
+    let state = getState();
+    if (state[name].stage === 'UNINITIALIZED' || ens === null) {
+      await dispatch(initializeEnsSlice());
+    }
+    state = getState();
+    if (
+      state[name].stage === 'NO_NETWORK_SUPPORT' &&
+      !(
+        isBurnAddress(ensName) === false &&
+        isValidHexAddress(ensName, { mixedCaseUseChecksum: true })
+      ) &&
+      !isHexString(ensName)
+    ) {
+      await dispatch(resetResolution());
+    } else {
+      const recipient = ensName.trim();
```

Comment:
Should we `trim` the `ensName` at the very top of the function?

Reply:
I'm not sure what value that serves as we don't use recipient until this block. we're mainly clearing out whitespace before performing the actual lookup.

Reply:
I think David is right - it would be better to trim this early. It would ensure that the `isBurnAddress` and `isHexString` and `isValidHexAddress` checks above ignore the whitespace as well.

But it's probably better as-is for this PR because this is how it worked before. It's a bit hard to tell because of how it's spread between different component methods, but in the old code it wasn't trimmed until this same point in the logic.

Maybe worth a TODO comment? :shrug:

Reply:
I don't think there are any ramifications of doing this now so I went ahead and trimmed it at the onset. I think I may have either been high on cough syrup or on my mobile device when reviewing this feedback because I didn't grasp that ensName === recipient and actually I now find it weird that it was ever called recipient. 

so now its `trimmedEnsName = ensName.trim()` and prcoessEnsError now expects 'ensName' instead of 'recipient' as a property.